notification-svc/controllers/reschedulingController.ts

```diff
@@ -1,13 +1,18 @@
 import { Response, Request } from 'express'
-import sendEmail from '../models/sendEmail'
 import { reschedulingEventEmitter } from '../utils/customEventEmitters'
+import sendEmail from '../services/sendEmail'
+import generateAndShareHash from '../utils/encryption'
 
 class reschedulingController {
     prompt(req: Request, res: Response) {
-        const { patient, appointment } = req.body
+        const { appointment } = req.body
         let isPending = true
-        const listenerId = `${appointment}_${patient}`
-        sendEmail(`Dear ${patient}, would you like to take an earlier appointment ${appointment}.`)
+        const listenerId = `${appointment}`
+
+        const hash = generateAndShareHash(appointment)
+        const emailType = 'rescheduling-prompt'
+        sendEmail(emailType, appointment, hash)
```

Comment:
Here we should receive an object with a hash and encryptionIV from generateAndShareHash. And pass it to the sendEmail function.